src/redux/books/bookSlice.jsx

```diff
@@ -1,50 +1,69 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const baseUrl = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/HTWbya7aDLtLFT33O8JP/books';
+
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  const time = Date.now();
+  const response = await axios.get(`${baseUrl}?time=${time}`);
+
+  const books = Object.entries(response.data).map((item) => ({
+    ...item[1][0],
+    item_id: item[0],
+  }));
+  return books;
+});
+
+export const uploadBook = createAsyncThunk(
+  'books/uploadBook',
+  async (formData) => {
+    await axios.post(baseUrl, formData);
+
+    return formData;
+  },
+);
+
+export const deleteBook = createAsyncThunk('books/delete', async (id) => {
+  // console.log(id);
+  const url = `${baseUrl}/${id}`;
+  // console.log(url)
+
+  try {
+    await axios.delete(url);
+  } catch (error) {
+    throw new Error();
+  }
+  return id;
+});
```

Comment:
- [ ] After deleting a book, Users need to refresh the page to see the changes but deleted book should remove from the display without refreshing the page. Please make sure to fix this issue. To fix this You should also delete that book from the local state so it will be removed from display immediately without refreshing the page.